src/UNL/UCBCN/Manager/Calendar.php

```diff
@@ -24,19 +26,53 @@ public function __construct($options = array()) {
         # to simply decide whether an event should be archived or posted based on its dates, instead
         # of a column that we set in the database
         $this->archiveEvents();
+
+        $allowed_tabs = array('pending', 'upcoming', 'past');
+        if (array_key_exists('tab', $_GET) && in_array($_GET['tab'], $allowed_tabs)) {
+            $this->tab = $_GET['tab'];
+        } else {
+            $this->tab = 'pending';
+        }
+
+        if (array_key_exists('page', $_GET) && is_numeric($_GET['page']) && $_GET['page'] >= 1) {
+            $this->page = $_GET['page'];
+        } else {
+            $this->page = 1;
+        }
     }
 
-    public function getCategorizedEvents() {
-        
+    public function getCategorizedEvents()
+    {
         $categories = array(
             'pending'  => $this->calendar->getEvents(CalendarModel::STATUS_PENDING),
             'posted'   => $this->calendar->getEvents(CalendarModel::STATUS_POSTED),
             'archived' => $this->calendar->getEvents(CalendarModel::STATUS_ARCHIVED)
         );
-        
+
         return $categories;
     }
 
+    public function getEvents() 
+    {
+        $events = NULL;
+        switch ($this->tab) {
+            case 'pending':
+                $events = $this->calendar->getEvents(CalendarModel::STATUS_PENDING, 10, ($this->page-1)*10);
```

Comment:
Will this break the last page of events if the total number is not divisible by 10?